src/mlpack/methods/ann/gan/gan_impl.hpp

```diff
@@ -72,7 +71,8 @@ GAN<Model, InitializationRuleType, Noise, PolicyType>::GAN(
   this->discriminator.predictors.set_size(predictors.n_rows,
       predictors.n_cols + batchSize);
   this->discriminator.predictors.cols(0, predictors.n_cols - 1) = predictors;
-
+  this->predictors = arma::mat(this->discriminator.predictors.memptr(),
```

Comment:
I think it should be the other way around. Here you have aliased the `Discriminator` predictors as `GAN` predictors.

Reply:
The discriminator predictors also include the fake data at the end.
So, we would have to do something like this
```
discriminator.predictors.cols(0, numsFunctions- 1) = arma::mat(predictors.memptr(), predictors.n_rows, predictors.n_cols, false, false);
```

I tried testing this. The data gets copied in the appropriate location in discriminator predictors but it doesn't seem like they are sharing memory with the GAN predictors. Maybe because we cannot have a larger matrix that shares memory with multiple different smaller matrix as that would require maintaining multiple pointers for the large matrix.
Do you have any different approach in mind?

Reply:
Try changing the size of the `GAN` predictors and responses, to the size of the `Discriminator` counterparts, and instead of directly copying into `discriminator.predictors` at places like this
```
discriminator.predictors.cols(numFunctions, numFunctions + batchSize - 1) =
      boost::apply_visitor(outputParameterVisitor, generator.network.back());
```
do
```
predictors.cols(numFunctions, numFunctions + batchSize - 1) =
      boost::apply_visitor(outputParameterVisitor, generator.network.back());
```